app/models/rm_api_customer_root.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class RmApiCustomerRoot < RmApiResource
+  request_body_type :json
+
+  get :all, '/'
+  put :update, '/'
+
+  def update_label(label_id, attrs)
+    # prune the root to not include labels with displayLabel=""
+    # because RM API would throw an error in PUT request
+    prune(label_id)
+    # patch in the new object here
+    patch(label_id, attrs)
+    # update the root
+    update
+  end
+
+  def prune(id)
+    labels.delete_if { |item| item.id != id && item.displayLabel == '' }
+  end
+
+  def patch(id, attrs)
+    labels.each do |label|
```

Comment:
this is acting on a singleton label right?  instead of the `each` you could probably do something like `label = labels.find { |label| label.id == id }` and then set your data attributes on that.  not sure if it's any more efficient (maybe even less efficient?) but i think it's easier to read.

Reply:
Sure thing. 
Regarding the usage of "each" vs. "find", I think each is more efficient for traversing an array and replacing content in place, "find" returns a new array with the replaced values.

Reply:
k,that makes sense 👍